### PR TITLE
fix(CodeTransform): smart select Java version

### PR DESCRIPTION
**Pull request by dhasani23**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Description

We want to be able to pre-select the Java version of the user's selected `pom.xml`, if we are able to successfully identify it. If we cannot identify it, we show some text saying that we were unable to detect your Java version, and still allow them to override us and select a Java version (8 or 11) and submit the job for transformation.

Smaller changes: missing an entry for the `build-logs.txt` file in our `manifest.json`, added `-q` to the `mvn` commands we run so that the `stdout` from them is more concise (`-q` = quiet), added the job status as a field to the `vcsViewerSubmitted` and `vcsViewerCanceled` metrics.

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [x] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt [hunk 1]
```diff
@@ -127,6 +131,17 @@ fun String.toTransformationLanguage() = when (this) {
     else -> TransformationLanguage.UNKNOWN_TO_SDK_VERSION
 }
 
+fun getJdkVersionText(version: JavaSdkVersion?): String {
+    val jdkVersionText: String = if (CodeModernizerUIConstants.supportedSourceJDKs.contains(version)) { // detected java version is supported
+        "We found Java version: $version. Select a different version if incorrect."
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt:136]: For the string internationalization, we have to use the `MessageBundle.properties` file to define strings.

You reference those strings like 
```message(<messagebundleId>)``` 

where the `messageBundleId` looks something like -> ```message("codemodernizer.notification.warn.zip_creation_failed.reasons.unknown")```

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt [hunk 2]
```diff
@@ -71,49 +74,61 @@ class PreCodeTransformUserDialog(
...
-                        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId()
+                        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     )
                 }
             }
+            row {
+                cell(jdkVersionLabel)
+            }
+            row {
+                javaInputSdkComboBox = comboBox(javaTransformInputSdks.map { it })
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt:122]: Do we need to map the selectedIndex like the above combo box 
`.bind({ it.selectedIndex }, { t, v -> t.selectedIndex = v }, model::focusedJavaInputIndex.toMutableProperty())`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt [hunk 3]
```diff
@@ -71,49 +74,61 @@ class PreCodeTransformUserDialog(
...
+                    .align(AlignX.FILL)
+                    .columns(COLUMNS_MEDIUM)
+                    .component
+                if (javaTransformInputSdks.contains(model.selectedJavaModuleVersion)) javaInputSdkComboBox.selectedItem = model.selectedJavaModuleVersion
+                javaInputSdkComboBox.whenItemSelected {
+                    dialogPanel.apply() // apply user changes to model
```
Comment (damntrecky) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt:128]: Do we need to set the selected model index here like `model.selectedJavaModuleVersion = javaTransformInputSdks[model.focusedJavaInputIndex]`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt (hunk 2 above)
Comment (dhasani23) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt:124]: Ah yes thank you. Now I understand the purpose of that. Did much more thorough manual testing this time around to ensure everything works properly.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt (hunk 3 above)
Comment (dhasani23) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt:131]: Same as above, fixed.

**Review by damntrecky — APPROVED** [review]

**Comment by dhasani23** [comment]
Screenshot 2024-02-13 at 1 05 49 PM
Screenshot 2024-02-13 at 1 06 30 PM
Screenshot 2024-02-13 at 1 07 09 PM

**Comment by dhasani23** [comment]
This ^ is what the new experience looks like when, from top to bottom: we find a supported Java version (it gets pre-selected), find an unsupported Java version, and when we find "null" / error out when trying to find the Java version.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt [hunk 4]
```diff
@@ -127,6 +132,17 @@ fun String.toTransformationLanguage() = when (this) {
...
+fun getJdkVersionText(version: JavaSdkVersion?): String {
+    val jdkVersionText: String = if (version == null) {
+        message("codemodernizer.customerselectiondialog.unknown_jdk")
+    } else if (CodeModernizerUIConstants.supportedSourceJDKs.contains(version)) { // detected java version is supported
+        message("codemodernizer.customerselectiondialog.found_supported_jdk", version)
+    } else {
+        message("codemodernizer.customerselectiondialog.found_unsupported_jdk", version) // found the version, but unsupported
+    }
+    return jdkVersionText
+}
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt:144]:
```suggestion
fun getJdkVersionText(version: JavaSdkVersion?): String =
    if (version == null) {
        message("codemodernizer.customerselectiondialog.unknown_jdk")
    } else if (CodeModernizerUIConstants.supportedSourceJDKs.contains(version)) { // detected java version is supported
        message("codemodernizer.customerselectiondialog.found_supported_jdk", version)
    } else {
        message("codemodernizer.customerselectiondialog.found_unsupported_jdk", version) // found the version, but unsupported
    }
```

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt (hunk 4 above)
Comment (dhasani23) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt:144]: Thanks. Will put this in the next PR.

**Pull request merged**